test/unit/test_unit_metobs.py

```diff
@@ -0,0 +1,691 @@
+            with pytest.raises(NotImplementedError):
+                client.get_stations()
+
+            with pytest.raises(NotImplementedError):
+                client.get_stations(parameter, parameter_title)
+        elif parameter:
+            client.get_stations(parameter)
+            assert client.station == expected_station
+        else:
+            client.get_stations(parameter_title=parameter_title)
+            assert client.station == expected_station
+
+        if parameter or parameter_title:
+            mock_metobsstationv1.assert_called_once()
+
+    @pytest.mark.parametrize(
+        "station, stationset, expected_period",
+        [(None, None, None), ("S1", None, "P1"), (None, "S2", "P2")],
+    )
+    @patch("smhi.metobs.MetObsPeriodV1", return_value=1)
+    def test_unit_metobs_get_periods(
+        self,
+        mock_metobsperiodv1,
+        station,
+        stationset,
+        expected_period,
+    ):
+        """
+        Unit test for MetObs get_stations method.
+
+        Args:
+            mock_metobsperiodv1: mock of MetObsPeriodV1
+            station: station of api
+            stationset: station set of api
+            expected_period: expected period
+        """
+        client = MetObs()
+        client.station = MagicMock()
+        mock_metobsperiodv1.return_value = expected_period
+
+        if station is None and stationset is None:
+            with pytest.raises(NotImplementedError):
+                client.get_periods()
+
+            with pytest.raises(NotImplementedError):
+                client.get_periods(station, stationset)
+        elif station:
+            client.get_periods(station)
+            assert client.period == expected_period
+        else:
+            client.get_periods(stationset=stationset)
+            assert client.period == expected_period
+
+        if station or stationset:
+            mock_metobsperiodv1.assert_called_once()
+
+    @patch("smhi.metobs.io.StringIO")
+    @patch("smhi.metobs.pd.read_csv")
+    @patch("smhi.metobs.MetObsDataV1")
+    def test_unit_metobs_get_data(
+        self, mock_metobsdatav1, mock_pd_read_csv, mock_stringio
+    ):
+        """
+        Unit test for MetObs get_data method.
+
+        Args:
+            mock_metobsdatav1: mock of metobs data class
+            mock_pd_read_csv: mock of pandas read_csv
+            mock_stringio: mock of io StringIO
+        """
+        client = MetObs()
+        client.period = MagicMock()
+        client.get_data()
+
+        assert client.data == mock_metobsdatav1.return_value
+        assert client.table_raw == mock_metobsdatav1.return_value.get()
+
+        mock_metobsdatav1.assert_called_once()
+        mock_pd_read_csv.assert_called_once()
+        mock_stringio.assert_called_once()
+
+    @patch("smhi.metobs.MetObs.get_parameters")
+    @patch("smhi.metobs.MetObs.get_stations")
+    @patch("smhi.metobs.MetObs.get_periods")
+    @patch("smhi.metobs.MetObs.get_data")
```

Comment:
Here, write something like
`@patch("smhi.metobs.MetObs.get_data", return_value=("test1", "test2"))`
OR
before line 217 do
`mock_get_data.return_value = ("test1, "test2")`

and check that `client.get_data_from_selection` actually returns that.

Reply:
Resolved in [ea44e0c](https://github.com/Ingenjorsarbete-For-Klimatet/ifk-smhi/pull/4/commits/ea44e0c76c9cdeaec92fde9a0d9b5903d213f9d7).